### PR TITLE
Flatpak scripts of Sideka

### PR DESCRIPTION
**Pull request by uajain**
Concerns:
1 ) https://github.com/uajain/sideka-flatpak/issues/1
2 ) Whether this should go on flathub ?
3 ) App-center integration (some issues around app-data, I think, currently taking a deeper look)

### CODE REVIEW

File: id.sideka.App.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- See https://wiki.gnome.org/GnomeGoals/AppDataGnomeSoftware -->
+<component type="desktop">
+  <id>id.sideka.App.desktop</id>
+  <metadata_licence>CC-0</metadata_licence>
+  <name>Sideka</name>
+  <summary>Village management application</summary>
+  <description>
+    <p>
+      platform bagi pemerintah desa untuk menata kelola desa, melakukan pelayanan publik, dan menyuarakan suara desa.
```
Comment (cosimoc) [inline id.sideka.App.appdata.xml]: Should probably be capitalized

File: id.sideka.App.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    </p>
+  </description>
+  <launchable type="desktop-id">id.sideka.App.desktop</launchable>
+  <screenshots>
+    <screenshot type="default">
+      <image type="source">sideka-screenshot.png</image>
```
Comment (cosimoc) [inline id.sideka.App.appdata.xml]: This requires a full online URL, not a local path. You can assume that this file will be available in the master branch here; see e.g. how we do it in eos-photos: https://github.com/endlessm/eos-photos/blob/master/data/com.endlessm.photos.appdata.xml#L52

File: id.sideka.App.json [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+					"dest": "main"
+				},
+				{
+					"type": "file",
+					"path": "yarnrc",
+					"dest": "main",
+					"dest-filename": ".yarnrc"
+				},
+				{
+				  "type": "file",
```
Comment (cosimoc) [inline id.sideka.App.json]: Nitpick: indentation for these is different compared to the rest of the file

File: id.sideka.App.json [hunk 4]
```diff
@@ -0,0 +1,153 @@
+{
+	"app-id": "id.sideka.App",
+	"runtime": "org.freedesktop.Platform",
+	"runtime-version": "1.6",
+	"base": "io.atom.electron.BaseApp",
+	"base-version": "stable",
+	"branch": "stable",
+	"sdk": "org.freedesktop.Sdk",
+	"command": "run.sh",
+	"separate-locales": false,
```
Comment (cosimoc) [inline id.sideka.App.json]: Why did you have to set this?

**Comment by cosimoc** [comment]
@uajain As for your question about Flathub, I think that would be a good idea, but let's first see if we can complete a build that also works with the app center infrastructure.

File: id.sideka.App.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- See https://wiki.gnome.org/GnomeGoals/AppDataGnomeSoftware -->
+<component type="desktop">
+  <id>id.sideka.App.desktop</id>
+  <metadata_licence>CC-0</metadata_licence>
```
Comment (nedrichards) [inline id.sideka.App.appdata.xml]: Needs a 
`<project_license>AGPL</project_license>
`
Element as well

**Comment by nedrichards** [comment]
I'd look at the localization of the text elements a little more - another example: https://github.com/flathub/org.kde.ktuberling/blob/master/org.kde.ktuberling.appdata.xml certainly I feel that we should have everything in Bahasa and "id" localization and ideally an english version as well.

Some questions I know Flathub reviewers will ask: 
- You need to try and send the desktop file, patch and appdata upstream
- You need a <versions> tag and ideally changelog
- Are upstream interested in having access to the flathub project? And helping to maintain?

File: id.sideka.App.json (hunk 4 above)
Comment (uajain) [reply id.sideka.App.json]: It came from the flatpak-builder-tools template.
See https://github.com/flatpak/flatpak-builder-tools/blob/master/yarn/io.atom.electron.ElectronQuickStart.json#L10
Should it be removed? Not relevant to this context?

File: id.sideka.App.json [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+				"generated-sources.json",
+				{
+					"type": "patch",
+					"path": "electron-chromedriver.patch",
+					"dest": "electron-chromedriver-1.7.1"
+				},
+				{
+					"type": "file",
+					"path": "yarn.lock",
+					"dest-filename": "yarn.lock",
```
Comment (cosimoc) [inline id.sideka.App.json]: No need to specify this again, by default the `dest-filename` is the same as the name of the file.

File: id.sideka.App.json [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+				{
+					"type": "file",
+					"path": "yarnrc",
+					"dest": "main",
+					"dest-filename": ".yarnrc"
+				},
+				{
+					"type": "file",
+					"path": "icon.128x128.png",
+					"dest-filename": "icon.128x128.png"
```
Comment (cosimoc) [inline id.sideka.App.json]: Ditto

File: id.sideka.App.json [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+				},
+				{
+					"type": "file",
+					"path": "icon.64x64.png",
+					"dest-filename": "icon.64x64.png"
```
Comment (cosimoc) [inline id.sideka.App.json]: Ditto

File: id.sideka.App.json [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+				},
+				{
+					"type": "file",
+					"path": "id.sideka.App.desktop",
+					"dest-filename": "id.sideka.App.desktop"
```
Comment (cosimoc) [inline id.sideka.App.json]: Ditto

File: id.sideka.App.json [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+				},
+				{
+					"type": "file",
+					"path": "id.sideka.App.appdata.xml",
+					"dest-filename": "id.sideka.App.appdata.xml"
```
Comment (cosimoc) [inline id.sideka.App.json]: Ditto

File: id.sideka.App.json [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+			],
+			"build-commands": [
+				"yarn cache clean",
+				"tar -cf electron-chromedriver-1.7.1/* | gzip -n > electron-chromedriver-1.7.1.tgz",
+				"cp electron-chromedriver-1.7.1.tgz yarn-mirror/electron-chromedriver-1.7.1.tgz",
+				"yarn install --pure-lockfile --non-interactive --offline --cwd /run/build/sideka-app/main",
+				"yarn --cwd /run/build/sideka-app/main run postinstall",
+				"yarn --cwd /run/build/sideka-app/main run build:prod",
+				"mkdir -p /app/main /app/bin",
+				"cp -ra main/* /app/main",
```
Comment (cosimoc) [inline id.sideka.App.json]: The `-a` flag implies `-r`, so you should not need to use both together

**Pull request merged**